indico/util/mdx_latex.py

```diff
@@ -149,6 +156,69 @@ def unescape_latex_entities(text):
     return out
 
 
+def latex_render_error(message):
+    return textwrap.dedent(r"""
+       \begin{tcolorbox}[width=\textwidth,colback=red!5!white,colframe=red!75!black,
+                         title={Indico rendering error}]
+          \begin{verbatim}
+            %s
+          \end{verbatim}
+       \end{tcolorbox}""" % message)
+
+
+def latex_render_image(src, alt, strict=False):
+    """
+    Generate LaTeX code that includes an arbitrary image from a URL.
+
+    This involves fetching the image from a web server and figuring out its
+    MIME type. A temporary file will be created, which is not immediately
+    deleted since it has to be included in the LaTeX code. It should be handled
+    by the enclosing code.
+
+    :param src: source URL of the image
+    :param alt: text to use as ``alt="..."``
+    :param strict: whether a faulty URL should break the whole process
+    :returns: a (latex_code, file_path) tuple, containing the LaTeX code and
+              path to the temporary image file.
+    """
+    try:
+        if urlparse(src).scheme not in ('http', 'https'):
+            raise ImageURLException("URL scheme not supported: {}".format(src))
+        else:
+            request = requests.get(src)
+            extension = None
+
+            if request.status_code != 200:
+                raise ImageURLException("[{}] Error fetching image".format(request.status_code))
+
+            if 'content-type' in request.headers and request.headers['content-type']:
+                extension = guess_extension(request.headers['content-type'])
+                # as incredible as it might seem, '.jpe' will be the answer in some Python environments
+                if extension == '.jpe':
+                    extension = '.jpg'
+            if not extension:
+                # assume PNG if no extension info
+                extension = "png"
+
+            __, tempfile_name = mkstemp(prefix='indico-latex-', suffix=extension)
```

Comment:
`NamedTemporaryFile`? Also, do we ever delete this file? If not maybe add something like this:

    delete_file.apply_async(args=[tempfile_name], countdown=3600)